app/components/CreateTask.js

```diff
@@ -1,32 +1,43 @@
 import React from 'react';
+import TasksManager from '../lib/tasksManager';
 
 export default class CreateTask extends React.Component {
 
   constructor(props) {
     super(props);
-    this.state = {rawString: ''};
+    this.state = {rawString: (this.props.unsavedTask || '')};
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return nextProps.unsavedTask !== this.state.rawString;
+  }
+
+  componentDidUpdate(props) {
+    this.setState({ rawString: props.unsavedTask })
   }
 
   onTextChange = (event) => {
+    // save this as the current 'unsaved' task if while we're not officially saving it as an actual task yet
     var rawString = event.target.value;
-    this.setState({rawString: rawString});
+    this.props.onUpdate(rawString);
   }
 
   handleKeyPress = (e) => {
     if (e.key === 'Enter') {
-      this.submitTask();
+      var rawString = event.target.value;
+      this.submitTask(rawString);
     }
   }
 
-  submitTask() {
-    this.props.onSubmit(this.state.rawString);
-    this.setState({rawString: ''})
+  submitTask(value) {
+    TasksManager.get().setUnsavedTask('');
```

Comment:
Why not call this in Tasks.js in `createTask`? Would remove the TasksManager dependency from this class.

Reply:
No worries, I just did to my fork of your PR. Just wanted to make sure I wasn't overlooking something obvious.